src/odemis/gui/comp/legend.py

```diff
@@ -538,45 +548,124 @@ def Refresh(self):
+        if self.legend_on:
+            self._callback(choices_radio[0])  # only call callback fct when change requested via legend
+
+        self.control_sizer.Add(self.pos_display, 2, border=10, flag=wx.ALIGN_CENTER | wx.RIGHT)
+
+        # refresh layout
+        self.border_sizer.Layout()
+        self.Parent.Layout()  # need to be called as method is called after self.Layout im main thread
 
-    def set_pol_callback(self, func):
+    def OnPositionButton(self, event):
+        """
+        In the case of an event (select a button), the radio button selection is
+        requested and the callback function, which changes the position is called.
+        The _callback function can be overwritten by the viewport.
+        """
+        pos = self.pos_display.GetValue()
+        self._callback(pos)
+
+    def set_callback(self, func):
+        """
+        Sets the callback function that is executed when a new radio button selection
+        is requested (in OnPositionButton).
+        :param func: (instancemethod) callback function that is called when a new pos is requested
+        """
         self._callback = func
 
-    def set_pol_value(self, pol):
-        self.pol_display.SetValue(pol)
+    def set_value(self, pos):
+        """
+        Sets the new position requested in the legend
+        :param pos: (str) new radio button selection to be set active
+        """
+        if self.pos_display_combo and pos not in self.pos_display.choices:
+            # only called when new pos is requested via stream settings panel not legend
+            self.legend_on = False
+            for item in self.choices_dict.values():
+                if pos in item:
+                    pos_combo = self.choices_dict.keys()[self.choices_dict.values().index(item)]
+                    self.pos_display_combo.SetValue(pos_combo)
+                    self.OnPositionCombo(True)  # create the corresponding radio buttons
+                    self.pos_display.SetValue(pos)  # overwrite default value with requested value
+                    self.legend_on = True
+                    break
+        self.pos_display.SetValue(pos)
 
     @call_in_wx_main
-    def set_pol_entries(self, choices, default):
+    def set_pos_entries(self, choices, default_value, name, choices_dict=None):
         """
-        Create radio buttons to switch between pol positions within legend.
-        If only one pol pos present in data, display pol pos as static text.
-        :param choices: polarization positions found in data
+        Create radio buttons to switch between positions within legend.
+        If only one pos present in data, display pos as static text.
+        :param choices: (frozenset) positions found in data
+        :param default_value: (str) default value that should be active
+        :param name: (str) text to be displayed describing the radio legend
+        :param choices_dict: (dict or None) If too many choices, it is possible to
```

Comment:
Would it be possible to only have the `choices` argument, and detect whether it's a set (in which case the function would behave as if `choices_dict` is None), or a dict (in which case it would behave as if `choices_dict` is passed, and choices would be the set of all the possible values). It'd avoid having to pass pretty much twice the same data.

Reply:
yes good idea! Done!